python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -570,47 +569,106 @@ def rolling(self, *args, **kwargs):
         """
         return cudf.core.window.rolling.RollingGroupby(self, *args, **kwargs)
 
+    def count(self, dropna=True):
```

Comment:
I'm +-1 on these changes where I don't think the previous method of dynamically adding attributes to reduce repetition was too bad. At the same time I agree this is easier to understand and lets us document aggregations better.

Another option is to use descriptors although I'd consider that even more magical:

```python
class GroupBy:
    sum = AggregationMethod("sum")  # adds a `sum` method and grabs docstring from `DataFrame.sum`?
    idxmin = AggregationMethod("idxmin")
    ...
```

Reply:
I think this is probably something to just get some opinions on. I try to avoid magic when it's not needed to make it easier for more devs, and here we also gain docs, but I'm fine with whatever solution we choose. Maybe I'll just post in Slack later and get some opinions.

Reply:
Let's go with this for now; it will be easier to go back to monkey-patching attributes if at some point we decide that's. what we want.

Reply:
Fine with me. Apologies for not posting on this earlier, last week seemed like a pretty busy time for everyone and there didn't seem to be much urgency to get this PR reviewed/merged, but now #7759 is blocked by this. I'll follow up with making this change if we decide to switch back.